File: app/src/main/java/com/example/android/moviesremake/retrofit/ApiInterface.java
package com.example.android.moviesremake.retrofit;

import com.example.android.moviesremake.retrofit.response.MoviesResponse;
import com.example.android.moviesremake.retrofit.response.ReviewResponse;
import com.example.android.moviesremake.retrofit.response.TrailerResponse;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * Created by PepovPC on 1/16/2018.
 */

public interface ApiInterface {
    @GET("movie/top_rated")
    Observable<MoviesResponse> getTopRatedMovies(@Query("api_key") String apiKey);

    @GET("movie/popular")
    Observable<MoviesResponse> getPopularMovies(@Query("api_key") String apiKey);

    @GET("movie/{id}/reviews")
    Observable<ReviewResponse> getMovieReviews(@Path("id") int id, @Query("api_key") String apiKey);

    @GET("movie/{id}/trailers")
    Observable<TrailerResponse> getMovieTrailers(@Path("id") int id, @Query("api_key") String apiKey);
}

File: app/src/main/java/com/example/android/moviesremake/retrofit/ApiInterface.kt
package com.example.android.moviesremake.retrofit

import com.example.android.moviesremake.retrofit.response.MoviesResponse
import com.example.android.moviesremake.retrofit.response.ReviewResponse
import com.example.android.moviesremake.retrofit.response.TrailerResponse

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * Created by PepovPC on 1/16/2018.
 */

interface ApiInterface {
    @GET("movie/top_rated")
    fun getTopRatedMovies(@Query("api_key") apiKey: String): Observable<MoviesResponse>

    @GET("movie/popular")
    fun getPopularMovies(@Query("api_key") apiKey: String): Observable<MoviesResponse>

    @GET("movie/{id}/reviews")
    fun getMovieReviews(@Path("id") id: Int, @Query("api_key") apiKey: String): Observable<ReviewResponse>

    @GET("movie/{id}/trailers")
    fun getMovieTrailers(@Path("id") id: Int, @Query("api_key") apiKey: String): Observable<TrailerResponse>
}
